components/funder/src/mutual_credit/incoming.rs

```diff
@@ -262,7 +262,11 @@ fn process_response_send_funds(
         .ok_or(ProcessOperationError::RequestDoesNotExist)?
         .clone();
 
-    let dest_public_key = &mutual_credit.state().idents.remote_public_key;
+    let dest_public_key = if pending_transaction.route.public_keys.is_empty() {
+        &mutual_credit.state().idents.remote_public_key
+    } else {
+        pending_transaction.route.public_keys.last().unwrap()
+    };
```

Comment:
@pzmarzly : I think we shouldn't need the `if` here. If we ever arrive at this point in the code and the route is of length zero, I think it's a bug. Can you check what happens if we remove this `if`? (Only take the last public key from the route).

The `If` clause is only required at the other case.

Reply:
```text
failures:
    handler::tests::pair_basic::test_handler_pair_basic
    tests::tests::test_funder_basic
    tests::tests::test_funder_forward_payment
```

Reply:
@pzmarzly : What did you try that produced these results? Did you try leaving only `pending_transaction.route.public_keys.last().unwrap()`? I think that if this doesn't work we might have some bug here that needs to be fixed. Could you check this?

If it still doesn't work I can take a look if you want. Thanks!

Reply:
[Here's](https://github.com/pzmarzly/offst/commit/6fca2ebd46834053cb03486adb5adb16991ea359) the change that fails. I don't feel like looking into it anytime soon, so feel free to take a look.

Reply:
@pzmarzly : Thanks! Will do.

Reply:
@pzmarzly : I just checked it, seems cool to me now.
In the case of incoming ResponseSendFunds, what I did not understand earlier is that the route of the remembered corresponding RequestSendFunds message does not contain the next public key (Because we remove it before queueing the RequestSendFunds). 

Therefore for example in the case of two nodes: `B -- C` where B sends a request and C returns a response, during handling the response B looks up the original request, and the original request contains an empty route, which is ok in this stage.

It seems like the `if` clause is the right thing to do there. I might add some comment later explaining why this is happening.